### PR TITLE
Game api

### PR DESCRIPTION
**Pull request by jaimedp**

### CODE REVIEW

File: src/managers/run-strategies/multiplayer-strategy.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        if (!curUserId) {
+            return dtd.reject({ statusCode: 400, error: 'We need an authenticated user to join a multiplayer world. (ERR: no userId in session)' }, session).promise();
+        }
+
+        var worldApi = new WorldApiAdapter(opt);
+
+        var restoreInitRun = function (worlds, msg, xhr) {
+            if (!worlds || !worlds.length) {
+                return dtd.reject({ statusCode: 404, error: 'The user is not in any world.' }, { options: opt, session: session, xhr: xhr });
```
Comment (narenranjit) [inline src/managers/run-strategies/multiplayer-strategy.js]: Maybe also include the missing userid in the message

File: src/managers/run-strategies/multiplayer-strategy.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        var opt = $.extend({
+            account: session.account,
+            project: session.project,
+            group: session.groupName
+        }, _pick(this.options, ['account', 'project', 'group']));
+
+        var dtd = $.Deferred();
+
+        if (!curUserId) {
+            return dtd.reject({ statusCode: 400, error: 'We need an authenticated user to join a multiplayer world. (ERR: no userId in session)' }, session).promise();
```
Comment (narenranjit) [inline src/managers/run-strategies/multiplayer-strategy.js]: Is the .promise() at the end required?

File: src/service/world-api-adapter.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+    if (serviceOptions.token) {
+        transportOptions.headers = {
+            'Authorization': 'Bearer ' + serviceOptions.token
+        };
+    }
+
+    var http = new TransportFactory(transportOptions);
+
+    var setFilterOrThrowError = function (options) {
```
Comment (narenranjit) [inline src/service/world-api-adapter.js]: The only 'filter' allowed here is the id right? Maybe make more specific? Or are there other filters allowed?

File: src/managers/run-strategies/multiplayer-strategy.js (hunk 1 above)
Comment (jaimedp) [inline src/managers/run-strategies/multiplayer-strategy.js]: The userId is actually part of the session object, that also includes the account/project/group

File: src/managers/run-strategies/multiplayer-strategy.js (hunk 2 above)
Comment (jaimedp) [inline src/managers/run-strategies/multiplayer-strategy.js]: This is the outermost return from getRun(), if not the client would have the deferred object, which I think should not leak out if the fn

**Pull request merged**